### PR TITLE
CLOUDP-261305: Extend split e2e tests

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

_Jira ticket:_ CLOUDP-261305

This PR does the following:
1. Updates the dev and prod files with latest specs + new version
2. Adds a test to check that splitting works and generates a valid spec for OASes that have external dependencies (api registry)
3. Dynamically get the versions to avoid having them hardcoded

Core changes in tools/cli/test/e2e/cli/split_test.go

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by andreaangiolillo** [comment]
PR `#100`!!! 💯  😮 🙀

File: tools/cli/test/e2e/cli/split_test.go [hunk 1]
```diff
@@ -2,27 +2,26 @@ package cli
...
 )
 
-var versions = []string{
-	// "2023-01-01",
-	// "2023-02-01",
-	"2023-10-01",
-	"2023-11-15",
-	"2024-05-30",
-	"2025-01-01",
+var skipVersions = []string{
```
Comment (blva) [inline tools/cli/test/e2e/cli/split_test.go:19]: still working on adding these two versions in a next PR

Comment (blva) [reply tools/cli/test/e2e/cli/split_test.go:19]: here https://github.com/mongodb/openapi/pull/153

File: tools/cli/test/e2e/cli/split_test.go [hunk 2]
```diff
@@ -75,19 +86,105 @@ func TestSplitVersions(t *testing.T) {
...
+	cmd.Stdout = &o
+	cmd.Stderr = &e
+	require.NoError(t, cmd.Run(), e.String())
+
+	versions := getVersions(t, cliPath, base, folder)
+	fmt.Printf("Versions: %v\n", versions)
+	for _, version := range versions {
+		if folder == "prod" && version == "2025-01-01" {
+			continue
+		}
```
Comment (andreaangiolillo) [inline tools/cli/test/e2e/cli/split_test.go:130]: can we change this to be `if "prod" and date in the future?`

**Review by andreaangiolillo — APPROVED** [review]

File: tools/cli/test/e2e/cli/split_test.go (hunk 2 above)
Comment (blva) [reply tools/cli/test/e2e/cli/split_test.go:130]: yes! good point

Comment (blva) [reply tools/cli/test/e2e/cli/split_test.go:130]: I'm opening another follow-up pr so I'll add this right away in https://github.com/mongodb/openapi/pull/153

**Pull request merged**